azure-pipelines.yml

```diff
@@ -159,50 +150,35 @@ stages:
           az configure --defaults acr="$ACR_NAME"
           az acr login
 
-    - bash: |
-        set -eux  # fail on error
-
-        # Build runtime Docker image
-        # Reuses the cached build stage from the previous docker build task
-        docker build --build-arg VersionPrefix="$SEMANTIC_VERSION" \
-          -t "$ACR_NAME.azurecr.io/k2bridge:$SEMANTIC_VERSION" \
-          .
-      displayName: Build Runtime Image
+          # Build runtime Docker image
+          # Reuses the cached build stage from the previous docker build task
+          docker build --build-arg VersionPrefix="$SEMANTIC_VERSION" \
+            -t "$ACR_NAME.azurecr.io/k2bridge:$SEMANTIC_VERSION" \
+            .
 
-    - task: AzureCLI@1
-      displayName: Push Image
-      inputs:
-        azureSubscription: $(ACR_PUSH_SERVICE_CONNECTION)
-        scriptLocation: inlineScript
-        inlineScript: |
-          set -eux  # fail on error
+          # Build end-to-end test image
+          docker build --target end2endtest \
+            --build-arg VersionPrefix="$SEMANTIC_VERSION" \
+            -t "$ACR_NAME.azurecr.io/k2bridge-test:$SEMANTIC_VERSION" \
+            .
 
-          # Push Docker image
+          # Push Docker image to ACR
           docker push "$ACR_NAME.azurecr.io/k2bridge:$SEMANTIC_VERSION"
+          docker push "$ACR_NAME.azurecr.io/k2bridge-test:$SEMANTIC_VERSION"
```

Comment:
This was originally removed from the "build" stage since it takes more time and delays the overall progress of this pipeline. Why have you returned this?